Gcp/secure source manager

sreedharashwin opened the pull request:
Raising pull request for GCP's Secure Source Manager service

Commit:
Google Secure Source Manager resource creation and relevant kms configuration check

Checking if the kms_key value is present in the resource creation plan or not. Key Management Service is used to encrypt resources or data. This can be managed by the customer instead of relying on the service provider

Commit:
Secure Source Manager instance creation with private configuration setting

private_config block checks if the given instance is private. By default, all instances are private as per the company policy.

Commit:
Terraform and policy files for ca_pool attribute

Terraform and policy files for ca_pool attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Updating policy file for kms_key attribute

Updating policy.rego for kms_key

Commit:
Updating policy file for private_config attribute

Updating policy.rego file for private_config attribute

Commit:
Terraform and policy files for workforce_identity_federation attribute

Files for workforce_identity_federation attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Terraform and policy files for IAM binding attribute

Terraform and policy files for binding attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Terraform and policy files for branch_rule attribute

Terraform and policy files for branch_rule attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Terraform and policy files for require_comments_resolved attribute

Terraform and policy files for require_comments_resolved attribute -

1. Includes compliant and non compliant terraform files
2. vars and policy rego files

Commit:
Terraform and policy files for linear_history attribute

Terraform and policy files for linear_history attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for require_pull_request attribute

Terraform and policy files for require_pull_request attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for reviews_count attribute

Terraform and policy files for reviews_count attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for default_branch attribute

Terraform and policy files for default_branch attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for description attribute

Terraform and policy files for description attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for git ignore attribute

Terraform and policy files for  git ignore attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for license attribute

Terraform and policy files for  license attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Terraform and policy files for readme attribute

Terraform and policy files for  readme attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Commit:
Adding read me file and the images used

Read me file to explain about the service and the resources available

Commit:
Terraform and policy files for policy_data attribute

Terraform and policy files for  policy_data attribute -

1. Includes compliant and non compliant terraform files
2. Policy rego file

Edited Read me file

sthenusan committed:
Gcp/service/scc (#47)

* Upgrade of helpers and new templates (#7)

+ Removal of AzureRM version to default to the latest version.

* add Init changes for scc-google_scc_event_threat_detection_custom_module-enablement_state

* add git ignore

* Add rough structures for other policies

* add changes in display_name

* add changes in display_name

* Add correct files for event-enablement_state

* add files for folder_custom_module

* Add files for organization custom module policy

* Add files for project custom  module policy

* Add changes for best practice in coding

* Add files for length check for display_name in source

* Add fixes for feedback

* Add changes for feedback of code review

---------

Co-authored-by: Pat <71543754+Pat-Megabite@users.noreply.github.com>

sthenusan committed:
Revert "Gcp/service/scc (#47)" (#48)

This reverts commit 1ad5709047e59b62ae3b6cb756e41285015bb2b0.

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/approval_count/policy.rego:
```diff
@@ -0,0 +1,18 @@
+conditions := [
+    [
+    {"situation_description" : "Enforce a minimum number of approvals",
+    "remedies":[ "Have at least 3 people approve requests"]},
```
Indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/approval_count/policy.rego:
Looks good, just fix up indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/comments/policy.rego:
```diff
@@ -0,0 +1,18 @@
+conditions := [
+    [
+    {"situation_description" : "Enforce a rule to require comments resolved before merging to the branch",
+    "remedies":[ "Have this enabled for the branch before merge"]},
```
Indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/comments/policy.rego:
Looks good just indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/linear_history/policy.rego:
```diff
@@ -0,0 +1,18 @@
+conditions := [
+    [
+    {"situation_description" : "Enforce a requirement for linear history before merging to the branch",
+    "remedies":[ "Have this enabled for easier branch management"]},
```
Indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/linear_history/policy.rego:
Looks good just indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/pull_request/policy.rego:
```diff
@@ -0,0 +1,18 @@
+conditions := [
+    [
+    {"situation_description" : "Enforce a pull request creation before merge",
+    "remedies":[ "Set the required flag for pull request"]},
```
Indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/pull_request/policy.rego:
```diff
@@ -0,0 +1,18 @@
+    {
+        "condition": "Checks if there is a pull request requirement before merge",
+        "attribute_path" : ["require_pull_request"],
+        "values" : [false],
```
probably better to just whitelist true

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/reviews_count/policy.rego:
```diff
@@ -0,0 +1,18 @@
+conditions := [
+    [
+    {"situation_description" : "Enforce a minimum number of reviews",
+    "remedies":[ "Have at least 2 people review requests"]},
```
Indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/branch_rule/reviews_count/policy.rego:
Looks good just indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/instance/ca_pool/policy.rego:
Remove template comments. Fix indentation. 
Potentially pattern whitelist is more appropriate but otherwise good.

Pat-Stuart commented on policies/gcp/securesourcemanager/instance/kms_key/policy.rego:
```diff
@@ -0,0 +1,20 @@
+conditions := [
+     [
+    {"situation_description" : "The Secure Source Manager instance must be configured to use a company-managed KMS key",
+    "remedies":[ "Review your Terraform configuration to reference the approved KMS key"]},
```
Indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/instance/private_config/policy.rego:
Looks good, just remove template comments and fix indentation.

Pat-Stuart commented on policies/gcp/securesourcemanager/instance/workforce_identity_federation/policy.rego:
Good, just indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/instance_iam/binding/policy.rego:
use instance_iam_binding as a folder and have this policy in it.

Fix indentation otherwise fine

Pat-Stuart commented on policies/gcp/securesourcemanager/policy/view_access/policy.rego:
```diff
@@ -0,0 +1,21 @@
+    "Double-check the policy configuration in your Terraform code."]},
+    {
+        "condition": "The policy must be only instanceAccessor'.",
+        "attribute_path" : ["policy_data"],
```
you should be able to access binding role through the attribute path rather than passing in an object

Pat-Stuart commented on policies/gcp/securesourcemanager/policy/view_access/policy.rego:
use instance_iam_policy as a separate folder and have this policy in it.

Pat-Stuart commented on policies/gcp/securesourcemanager/policy/view_access/policy.rego:
```diff
@@ -0,0 +1,21 @@
+    [
+    {"situation_description" : "Ensure the policy is correctly set.",
+    "remedies":[ "Set the role to instanceAccessor only",
+    "Double-check the policy configuration in your Terraform code."]},
```
Indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/repository/default_branch/policy.rego:
Not really security related, can probably remove

Pat-Stuart commented on policies/gcp/securesourcemanager/repository/description/policy.rego:
Good practice but probably not security related

Pat-Stuart commented on policies/gcp/securesourcemanager/repository/gitignore/policy.rego:
Good policy, ensuring relevant gitignores can prevent secret exposure.

Just fix up indentation

Pat-Stuart commented on policies/gcp/securesourcemanager/repository/license/policy.rego:
This policy probably needs review by @JBarazani for security relevance. We also might want to use licenses other than open source

Indentation needs to be fixed up. License list breaks template conventions but I understand why, I'll defer that decision.

Pat-Stuart commented on policies/gcp/securesourcemanager/repository/readme/policy.rego:
Good practice but potentially not security related

Pat-Stuart reviewed (changes requested):
Fix up service name to secure_source_manager. Use the full name of the resources for folders. We only want a folder depth of 3 for each service ie. service - resource - policy

I appreciate the test images and readme but we don't want to merge this so please remove.

Good security policies for the most part, good job.
Thanks for your contribution

Commit:
Fixing issues raised in the pull request.

1. Renamed the folders to match with the resources names
2. Removed the images folder and readme file
3. Fixed the indentation by running opa fmt
4. Removed policies that are not aligned with security (readme, description, default branch rule)
5. Fixed a policy to use pattern whitelist

Commit:
Merge branch 'gcp/secure-source-manager' of https://github.com/Hardhat-Enterprises/Policy-Deployment-Engine into gcp/secure-source-manager

Pat-Stuart commented on policies/gcp/secure_source_manager/google_secure_source_manager_instance_iam_policy/view_access/policy.rego:
This has the same issue of passing in a whole object when you should be able to just access the role.
Let me know if you are having issues with that.

Its not necessarily wrong its just not what we want to do.

Pat-Stuart reviewed (commented):
Looks good, if you could just address the IAM policy.

Yotam can make a decision on the license policy. 
Thanks for your time addressing feedback.

Commit:
Removing plan.json files

Parsing JSON object for a whitelisting policy

sreedharashwin commented:
Made the changes as suggested.

Pat-Stuart commented:
Closing for small changes still needed
